frontend/packages/console-app/src/components/cloud-shell/MultiTabbedTerminal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { Tabs, Tab, TabTitleText, TabTitleIcon, Tooltip } from '@patternfly/react-core';
+import { CloseIcon, PlusIcon } from '@patternfly/react-icons';
+import { useTranslation } from 'react-i18next';
+import CloudShellTerminal from './CloudShellTerminal';
+import './MultiTabbedTerminal.scss';
+
+interface TerminalData {
+  terminalNumber: number;
+}
+const MAX_TERMINAL_TABS = 8;
+
+export const MultiTabbedTerminal: React.FC = () => {
+  const [terminalTabs, setTerminalTabs] = React.useState<TerminalData[]>([{ terminalNumber: 1 }]);
+  const [activeTabKey, setActiveTabKey] = React.useState<number>(0);
+  const { t } = useTranslation();
+
+  const addNewTerminal = () => {
+    if (terminalTabs.length < MAX_TERMINAL_TABS) {
+      const tabs = [...terminalTabs];
+      const newTerminalNumber = terminalTabs[terminalTabs.length - 1].terminalNumber + 1;
+      tabs.push({
+        terminalNumber: newTerminalNumber,
+      });
+      setTerminalTabs(tabs);
+      setActiveTabKey(terminalTabs.length);
+    }
+  };
+
+  const removeCurrentTerminal = async (tabIndex) => {
+    const tabs = [...terminalTabs];
+    tabs.splice(tabIndex, 1);
+    setTerminalTabs(tabs);
+    setActiveTabKey(tabIndex > 0 ? tabIndex - 1 : terminalTabs.length - 2);
+  };
+
+  return (
+    <Tabs activeKey={activeTabKey} isBox>
+      {terminalTabs.map((tab, tabIndex) => (
+        <Tab
+          translate="no"
+          eventKey={tabIndex}
+          className="co-multi-tabbed-terminal__tab"
+          key={tab.terminalNumber}
+          onClick={() => setActiveTabKey(tabIndex)}
+          title={
+            <div>
+              <TabTitleText>
+                {t('console-app~Terminal {{number}}', { number: tabIndex + 1 })}
+              </TabTitleText>
+              <TabTitleIcon>
+                {terminalTabs.length > 1 ? (
+                  <CloseIcon
+                    data-test="close-terminal-icon"
+                    onClick={() => removeCurrentTerminal(tabIndex)}
+                  />
+                ) : (
+                  <Tooltip content={t('console-app~Cannot close last tab')}>
+                    <CloseIcon data-test="close-terminal-disabled-icon" disabled />
+                  </Tooltip>
+                )}
+              </TabTitleIcon>
+            </div>
+          }
+        >
+          <CloudShellTerminal />
+        </Tab>
+      ))}
+      <Tab
+        translate="no"
+        eventKey="add-tab"
+        key="add-tab"
+        onClick={() => terminalTabs.length < MAX_TERMINAL_TABS && addNewTerminal()}
+        title={
+          <TabTitleIcon>
+            {terminalTabs.length < MAX_TERMINAL_TABS ? (
+              <PlusIcon data-test="add-terminal-icon" />
+            ) : (
+              <Tooltip content={t('console-app~Cannot add any more tabs')}>
+                <PlusIcon data-test="add-terminal-disabled-icon" disabled />
+              </Tooltip>
+            )}
```

Comment:
Does UX want this button present if we hit the max number of terminals?

Was it discussed that there's a reason for having a maximum number of terminals? I can easily circumvent this maximum by opening a new browser window.
I'm ok with a maximum but just curious who defined it.

Reply:
No, the Max_Number is not defined, but as the Tab Component is actually very static the app starts to get slower when 10+ tabs are rendered. The max can be reset or removed.
Please suggest.

Reply:
PM agreed to the 8 terminal max.

cc @beaumorley 
I think the message should be more focused on the reason why we cannot add more tabs:
`Maximum number of sessions open`

Maybe `sessions` is bad because we can simply open a new window and create more sessions but it sounds better than using the word `tabs`.